src/gmock-win32.cpp

```diff
@@ -8,73 +8,120 @@
 #pragma warning(pop)
 
 #include <memory>
+#include <string>
 #include <stdexcept>
 
 #pragma comment(lib, "dbghelp.lib")
 
-#define RETURN_IF_FAILED(hr)            \
-do {                                    \
-    const auto hres = HRESULT{ hr };    \
-    if (FAILED(hres))                   \
-        return hres;                    \
+#define THROW_HRESULT(hr, text)                     \
+    throw std::runtime_error{                       \
+        std::string{ "message: " } + text +         \
+        "; HRESULT code: " + std::to_string(hr) };           
+
+#define THROW_ERROR_CODE(err, text)                 \
+    throw std::runtime_error{                       \
+        std::string{ "message: " } + text +         \
+        "; error code: " + std::to_string(err) };   
+
+#define RETURN_IF_FAILED_HRESULT(hr)                \
+do                                                  \
+{                                                   \
+    const auto hRes = HRESULT{ hr };                \
+    if (FAILED(hRes))                               \
+        return hRes;                                \
 } while (false)
 
+#define THROW_IF_FAILED_HRESULT(hr, text)           \
+do                                                  \
+{                                                   \
+    const auto hRes = HRESULT{ hr };                \
+    if (FAILED(hRes))                               \
+        THROW_HRESULT(hRes, text);                  \
+} while (false)   
+
 namespace {
+namespace module {
+
+    constexpr auto kernel32 = L"kernel32";
+    constexpr auto dbghelp  = L"dbghelp";
+
+} // namespace module
 
-	typedef DWORD (WINAPI *pfnGetLastError_t)( VOID );
-	pfnGetLastError_t g_pfnGetLastError = nullptr;
+    using UniqueHMODULE = std::unique_ptr<
+        std::remove_pointer_t< HMODULE >, BOOL(WINAPI*)(HMODULE) >;
 
-	typedef HANDLE (WINAPI *pfnGetCurrentProcess_t)( VOID );
-	pfnGetCurrentProcess_t g_pfnGetCurrentProcess = nullptr;
+namespace utils {
+
+    UniqueHMODULE loadModule(const wchar_t* moduleName)
+    {
+        if (const auto hmodule = ::LoadLibraryW(moduleName))
+            return UniqueHMODULE{ hmodule, &FreeLibrary };
+        else
+            THROW_ERROR_CODE(::GetLastError(), "gmock_win32 initialization failed");
 
-	typedef BOOL (WINAPI *pfnWriteProcessMemory_t)(HANDLE hProcess, LPVOID lpBaseAddress, LPCVOID lpBuffer, SIZE_T nSize, SIZE_T* lpNumberOfBytesWritten);
-	pfnWriteProcessMemory_t g_pfnWriteProcessMemory = nullptr;
+        return UniqueHMODULE{ nullptr, { } };
```

Comment:
Understand. Improved structuring. And statistical analyzers will not complain about resource leaks.